dlk/python/dlk/templates/include/matrix/shift_add.h

```diff
@@ -75,23 +75,25 @@ inline int calc_offset(int i, int w) {
 template<typename T>
 void matrix_shift_add(MatrixView<T, MatrixOrder::ColMajor>& buf,
                       MatrixView<T, MatrixOrder::ColMajor>& result,
-                      const struct convolution_parameters& p) {
+                      const struct convolution_parameters& p,
+                      const int block_offset) {
   Measurement::Start("matrix_shift_add1");
 
   const int h = p.input_height;
   const int w = p.input_width;
   const int oc = p.output_channels;
   const int kh = p.kernel_height;
   const int kw = p.kernel_width;
+  const auto col_block = buf.cols();
 
   // only 3x3 kernel is supported.
   assert(kh == 3 && kw == 3);
 
-  for (unsigned int j = 0; j < buf.cols(); ++j) {
+  for (unsigned int j = 0; j < col_block; ++j) {
```

Comment:
I'm not sure whether this change is necessary. It looks that c++ compiler can do inlining automatically.